packages/vector_graphics/lib/vector_graphics.dart

```diff
@@ -260,8 +260,52 @@ class _RenderVectorGraphics extends RenderProxyBox {
     markNeedsPaint();
   }
 
+  @override
+  bool hitTestSelf(Offset position) => true;
+
+  @override
+  bool get sizedByParent => true;
+
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    return constraints.smallest;
+  }
+
+  @override
+  bool get isRepaintBoundary => true;
+
+  final Matrix4 transform = Matrix4.identity();
+
   @override
   void paint(PaintingContext context, ui.Offset offset) {
+    if (_scaleCanvasToViewBox(transform, size, pictureInfo.size)) {
+      context.canvas.transform(transform.storage);
+    }
     context.canvas.drawPicture(_pictureInfo.picture);
   }
 }
+
+bool _scaleCanvasToViewBox(
+  Matrix4 matrix,
+  Size desiredSize,
+  Size pictureSize,
+) {
+  if (desiredSize == pictureSize) {
+    return false;
+  }
+  final double scale = math.min(
+    desiredSize.width / pictureSize.width,
+    desiredSize.height / pictureSize.height,
+  );
+  final Size scaledHalfViewBoxSize = pictureSize * scale / 2.0;
+  final Size halfDesiredSize = desiredSize / 2.0;
+  final Offset shift = Offset(
+    halfDesiredSize.width - scaledHalfViewBoxSize.width,
+    halfDesiredSize.height - scaledHalfViewBoxSize.height,
+  );
+  matrix
+    ..setIdentity()
```

Comment:
set identity to allow continually recycling of the matrix

Reply:
If we're doing this can we maybe just reuse the matrix if it's already set up right (i.e. we're scaling/translating it to the same amount)?

Reply:
ahh right, I can remove that